### PR TITLE
boards: arm: mps2_an521: Resize _remote and _ns

### LINKED ISSUE #52049: Update mps2_an521_remote for compatibility with mps2_an521_ns
`mps2_an521_remote` was added primarilly as a test platform for the FPU in CI, using the second M33 core on the AN521 since the primary core is missing FPU support.

There are a number of problems with this target, however, such as the memory map overlapping with `mps2_an521_ns`, meaning you can't use TF-M and the remote core at the same time.

The `_remote` target should be updated to have the base image located at `0x0038 B000` with 468 KB flash memory allocated, which will remain compatible with the current AN521 flash layout from TF-M: https://git.trustedfirmware.org/TF-M/trusted-firmware-m.git/tree/platform/ext/target/arm/mps2/an521/partition/flash_layout.h

**Additional context**

This platform makes a useful target for testing system device tree. Any changes made to the remote target should keep that goal in mind.

### PR DESCRIPTION
**Pull request by microbuilder**
The `mps2_an521_remote` and `mps2_an521_ns` targets have the same memory layout for code and ram, meaning that you can't use TF-M (`mps2_an521_ns`) and the second core (`mps2_an521_remote`) at the same time.

This commit updates the memory map of the `_ns` build targets as follows:

- Reduces the code memory region from 1 MB to 512 KB, maintaining the existing base memory address of `0x00100000`
- Maintains the existing 512 KB ram memory at `0x28100000`

It updates the `_remote` target as follows:

- Reserves 468 KB code memory at address `0x0038B000`
- Reserves 512 KB ram memory at address `0x28180000`

This ensures that the code region for the `_remote` target doesn't overlap with the code region used by the single flash image layout defined upstream in TF-M, which the `_ns` target is based upon.

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>

Fixes #52049

### CODE REVIEW

**Comment by microbuilder** [comment]
Looks like `_remote` code execution fails unless the code region is set to `0x100000`, which squashes the TF-M image if it was present:

```
code: memory@100000 {
	reg = <0x00100000 DT_SIZE_K(512)>;
};
```

File: soc/arm/arm/mps2/soc.c [hunk 1]
```diff
@@ -37,7 +37,7 @@ FPGAIO_INIT(2);
 /* The base address that the application image will start at on the secondary
  * (non-TrustZone) Cortex-M33 mcu.
  */
```
Comment (mbolivar-nordic) [inline soc/arm/arm/mps2/soc.c:39]: Ruh roh. Not in scope for this PR, but this really ought to come from the DT, it looks like.

**Review by mbolivar-nordic — APPROVED** [review]

File: soc/arm/arm/mps2/soc.c (hunk 1 above)
Comment (microbuilder) [reply soc/arm/arm/mps2/soc.c:39]: I agree, but this needs further rework beyond just the memory map and pulling values in from DT. I'll put together a more complete PR once I can allocate more time to it.

**Review by uLipe — APPROVED** [review]

File: soc/arm/arm/mps2/soc.c (hunk 1 above)
Comment (microbuilder) [reply soc/arm/arm/mps2/soc.c:39]: @mbolivar-nordic This is already causing issues. Can you make a suggestion on the 'correct' was to get `CPU1_FLASH_ADDRESS` from `code: memory@38B000` in the `_remote` target?

**Pull request merged**